Prox/Prox/Data/PlacesResult.swift

```diff
@@ -9,43 +9,43 @@ private enum DeferredResult {
     case failure
 }
 
-class PlacesResult {
+class DatabaseResult<T> {
```

Comment:
nit: -> `DeferredResult` or `DeferredResultWrapper`? I think we can use this for more than just Database results.